plugins/workspace-minimap/src/minimap.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A minimap is a miniature version of your blocks that
+ * appears on top of your main workspace. This gives you an overview
+ * of what your code looks like, and how it is organized.
+ * @author cesarades@google.com (Cesar Ades)
+ */
+
+import * as Blockly from 'blockly/core';
+
+// Events that should be send over to the minimap from the primary workspace
+const BLOCK_EVENTS_ = [
+    Blockly.Events.BLOCK_CHANGE,
+    Blockly.Events.BLOCK_CREATE,
+    Blockly.Events.BLOCK_DELETE,
+    Blockly.Events.BLOCK_DRAG,
+    Blockly.Events.BLOCK_MOVE];
+
+/**
+ * A minimap is a miniature version of your blocks that appears on
+ * top of your main workspace. This gives you an overview of what
+ * your code looks like, and how it is organized.
+ * @param {!Blockly.WorkspaceSvg} workspace The workspace to sit in.
+ */
+export class Minimap {
+    protected workspace_: Blockly.WorkspaceSvg;
+    protected minimap_: Blockly.WorkspaceSvg;
+    /**
+     * Constructor for a minimap
+     * @param {!Blockly.WorkspaceSvg} workspace The workspace to sit in.
+     */
+    constructor(workspace: Blockly.WorkspaceSvg) {
+        this.workspace_ = workspace;
+        const options = {
+            readOnly: true,
+        }
+        this.minimap_ = Blockly.inject('minimapDiv', options);
+        this.mirror = this.mirror.bind(this);
+    }
+  
+    /**
+     * Initialize.
+     */
+    init(): void {
+        this.workspace_.addChangeListener(this.mirror);
+    }
+
+    /**
+     * Creates the mirroring between workspaces. Passes on all desired events
+     * to the minimap from the primary worskpace.
+     * @param {Blockly.Events.Abstract} event The primary workspace event.
+     * @protected
```

Comment:
I would change this to `private mirror` unless you know of a use case for overriding this method! If it's `protected` that means we can never change it unless it's a binary breaking change (i.e. `protected` is equivalent to `public` from an API design standpoint).

So in the interest of keeping the API you have to maintain as small as possible, it's preferable to use `private` when possible!

Reply:
For ts it should be:

```ts
private mirror(event: Blockly.Events.Abstract): void { /* etc */ }
```

instead of using the `@private` annotation! Typescript has [actual access modifiers](https://www.typescripttutorial.net/typescript-tutorial/typescript-access-modifiers/) :D